xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp

```diff
@@ -849,6 +855,18 @@ bool CMMALVideo::GetPicture(DVDVideoPicture* pDvdVideoPicture)
     pDvdVideoPicture->iFlags  = DVP_FLAG_ALLOCATED;
     if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_FLAG_USER3)
       pDvdVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_INTERLACED)
+    {
+      pDvdVideoPicture->iFlags |= DVP_FLAG_INTERLACED;
+      if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_TOP_FIELD_FIRST)
+        pDvdVideoPicture->iFlags |= DVP_FLAG_TOP_FIELD_FIRST;
+      if (!firstInterlacedFrame)
+      {
+        pDvdVideoPicture->dts = DVD_NOPTS_VALUE;
+        pDvdVideoPicture->pts = DVD_NOPTS_VALUE;
```

Comment:
That is a mpeg2 sample. I get one packet for every field on my h264 channels.

15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1006844.555555, pts: 1046844.555556
15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1026844.555556, pts: 1226844.555556
15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1046844.555556, pts: 1246844.555557
15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1066844.555557, pts: 1146844.555555
15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1086844.555557, pts: 1166844.555555
15:59:34 T:123145327652864  NOTICE:  ----------- dts: 1106844.555557, pts: 1106844.555557